File: 3-Business/3-Modules/lm.Comol.Core.BaseModules/Repository/Presentation/View/Controls/IViewMultipleFileUploader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using lm.Comol.Core.DomainModel;

namespace lm.Comol.Core.BaseModules.Repository.Presentation
{
    public interface IViewMultipleFileUploader  : lm.Comol.Core.DomainModel.Common.iDomainView 
    {
        Int32 MaxFileInputsCount { get; set; }
        long IdFolder{ get; set; }
        Int32 IdCommunityRepository { get; set; }

        String Portalname {get;}
        String BaseFolder {get;}

        String CommunityName {set;}
        String FilePath {set;}
        Boolean AllowFolderChange { set; }
        Boolean VisibleToDonwloaders { get; set; }
        Boolean DownlodableByCommunity { get; set; }
        Boolean AllowPersonalPermission { get; set; }
        CoreModuleRepository Permission { get; set; }

        void InitializeControl(long idFolder, Int32 idCommunity,CoreModuleRepository permission);
        void LoadFolderSelector(long idFolder, Int32 idCommunity, Boolean showHiddenItems, Boolean forAdminPurpose);
    }
}

File: 3-Business/3-Modules/lm.Comol.Modules.Base/Presentation/CommunityFile/View/IviewMultipleFileUploader.vb


Imports lm.Comol.Core.DomainModel.Common
Imports lm.Comol.Modules.Base.DomainModel
Imports COL_BusinessLogic_v2
Imports lm.Comol.Core.DomainModel

Namespace lm.Comol.Modules.Base.Presentation
    Public Interface IviewMultipleFileUploader
        Inherits lm.Comol.Core.DomainModel.Common.iDomainView
        Property MaxFileInputsCount() As Integer
        Property FolderID() As Long
        Property RepositoryCommunityID() As Integer
        WriteOnly Property CommunityName() As String
        WriteOnly Property FilePath() As String
        WriteOnly Property AllowFolderChange() As Boolean
        Property VisibleToDonwloaders() As Boolean
        Property DownlodableByCommunity() As Boolean
        Property AllowPersonalPermission() As Boolean
        ReadOnly Property Portalname() As String
        ReadOnly Property BaseFolder() As String

        Sub InitializeControl(ByVal FolderID As Long, ByVal CommunityID As Integer, ByVal oPermission As ModuleCommunityRepository)
        Sub LoadFolderSelector(ByVal FolderID As Long, ByVal CommunityID As Integer, ByVal ShowHiddenItems As Boolean, ByVal AdminPurpose As Boolean)

    End Interface
End Namespace